Docs: Fix typo and clarify introduction in Functional API section (#5157)
google/flax: docs_nnx/nnx_basics.md

The Flax NNX Functional API establishes a clear boundary between reference/object semantics and value/pytree semantics. It also allows the same amount of fine-grained control over the state that Flax Linen and Haiku users are used to. The Flax NNX Functional API consists of three basic methods:  `nnx.split`, `nnx.merge`, and `nnx.update`.

> Below, we define a `StatefulLinear` `Module`. In the following sections, we will use this module to demonstrate the Functional API methods (`nnx.split`, `nnx.merge`, and `nnx.update`). It contains:

Reviewer: Let's just fix the typo:
```suggestion
Below is an example of `StatefulLinear` `Module` that uses the Functional API. It contains:
```
